### ISSUE TITLE
High speed cameras generate warning about unused codec option

### ISSUE THREAD

**Issue by alexpiet** [AllenNeuralDynamics/dynamic-foraging-task#588]
02:10:24 PM:INFO:Foraging:Foraging.py:log_subprocess_output:line 4288:BONSAI: [out#0/avi @ 0000013ff6d2af00] Codec AVOption crf (Select the quality for constant quality mode) has not been used for any stream. The most likely reason is either wrong type (e.g. a video option with no video streams) or that it is a private option of some encoder which was not actually used for any stream.

I checked, and the logs include this warning before the recent addition of the `-v 24` verbose option, it was just getting buried in other information. This message appears twice, once for each camera. It appears whether I pressed "preview" or "record"

This is the codec in `Settings_box1.csv` 
`codec,-vcodec h264_nvenc -crf 23 -preset fast -b:v 50M`

I dont know what `-crf 23` does, or if it matters. @XX-Yin @ZhixiaoSu

**Comment by alexpiet** [comment]
@ZhixiaoSu @XX-Yin Can you investigate this?

**Comment by ZhixiaoSu** [comment]
In cpu compression, -crf 23 controls the compression level, the larger the number, the higher the compression. But as we switched to gpu compression, we should change the flag to -rc vbr -cq 23.

**Comment by ZhixiaoSu** [comment]
I attempted to test different codecs, but the codec in video writer doesn't seem to update according to the one in box settings. @XX-Yin Is there a different location I should check to update the codec?

**Comment by XX-Yin** [comment]
There are four boxes. I assume that you changed the right Settings_box#.csv?

**Comment by alexpiet** [comment]
Might be obvious - but you need to close the GUI after changing the codec, you can't just start/stop the camera.

**Comment by ZhixiaoSu** [comment]
@XX-Yin Yes, I change the one in setting folder. For 446, each computer only has one settings file, so it's hard to be wrong. And, yes, @alexpiet I restarted the GUI together with Bonsai and check the codecs after restarting. The parameter I tested with was bitrate, changing the number to 25M didn't change the one in bonsai.

**Comment by ZhixiaoSu** [comment]
I just figured out why: since the cameras are in restart loggers, the codec doesn’t update until we trigger the camera. After triggering, the codec get updated.

**Comment by XX-Yin** [comment]
[like]  Xinxin Yin reacted to your message:
________________________________

**Comment by ZhixiaoSu** [comment]
I tested '-rc vbr -cq 23' on 6A and 6B, it worked well without warnings. I'll check the video quality of 6A, which still has this codec now. If it looks ok, let's update all the codecs.

**Comment by XX-Yin** [comment]
Hi @ZhixiaoSu, looks great! Can you also check if the video quality/size changes with the `-cq value`?

**Comment by ZhixiaoSu** [comment]
It should in theory, as it has similar functions as -crf in cpu codec. Let me check it with a real mouse there.

**Comment by ZhixiaoSu** [comment]
Videos look ok, @mlee10201 will help update the codecs this week.

**Comment by alexpiet** [comment]
We should make a PR to change the settings template file so all the rigs use the same codec. @mlee10201 share the final codec you used.

**Comment by galenlynch** [comment]
Is this the same codec as the one Bruno was recommending?

**Comment by mlee10201** [comment]
codec,-vcodec h264_nvenc -rc vbr -cq 23 -preset fast -b:v 50M -v 24
this is what it's been updated to

**Comment by ZhixiaoSu** [comment]
Let’s also include sound attenuation in the settings. I can add it this week.
